confidant/authnz/__init__.py

```diff
@@ -237,6 +247,51 @@ def decorated(*args, **kwargs):
     return decorated
 
 
+def require_role(role):
+    """
+    Get the users role in credential metadata
+    """
+    def decorated(f):
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            cred_id = kwargs['id']
+            cred = Credential.get(cred_id)
```

Comment:
You should defer getting the credential until necessary, otherwise it adds load on the system, even when USE_ROLES is disabled, for example

Reply:
Or, as mentioned in another comment, change this to a function (not a decorator) which takes metadata as an argument, which will let you avoid doing a get on the credential.